src/app/components/applications/SSV/RegisterValidatorHome/components/GenerateKeyShares/GenerateKeyShares.tsx

```diff
@@ -2,112 +2,157 @@ import React, { useEffect } from 'react';
-import config from '~app/common/config';
 import { useStores } from '~app/hooks/useStores';
 import BorderScreen from '~app/components/common/BorderScreen';
 import HeaderSubHeader from '~app/components/common/HeaderSubHeader';
 import SecondaryButton from '~app/components/common/Button/SecondaryButton';
 import WalletStore from '~app/common/stores/applications/SsvWeb/Wallet.store';
 import AccountStore from '~app/common/stores/applications/SsvWeb/Account.store';
+import ProcessStore from '~app/common/stores/applications/SsvWeb/Process.store';
+import { getStoredNetwork, NETWORKS } from '~root/providers/networkInfo.provider';
 import ValidatorStore from '~app/common/stores/applications/SsvWeb/Validator.store';
 import NewWhiteWrapper from '~app/components/common/NewWhiteWrapper/NewWhiteWrapper';
-import ProcessStore, { RegisterValidator, SingleCluster } from '~app/common/stores/applications/SsvWeb/Process.store';
-import { useStyles } from '~app/components/applications/SSV/RegisterValidatorHome/components/GenerateKeyShares/GenerateKeyShares.styles';
-import { getStoredNetwork } from '~root/providers/networkInfo.provider';
+import useValidatorRegistrationFlow, { EValidatorFlowAction } from '~app/hooks/useValidatorRegistrationFlow';
+import {
+  useStyles,
+} from '~app/components/applications/SSV/RegisterValidatorHome/components/GenerateKeyShares/GenerateKeyShares.styles';
+import { translations } from '~app/common/config';
+
+type ButtonData = {
+  isShow: boolean
+  subText?: string
+  wrapperProps: {
+    className: string,
+  },
+  buttonProps: {
+    dataTestId: string,
+    withVerifyConnection?: boolean,
+    text: string,
+    withoutBackgroundColor?: boolean,
+    noCamelCase?: boolean,
+    submitFunction: Function,
+  },
+};
 
 const GenerateKeyShares = () => {
-  const stores = useStores();
-  const navigate = useNavigate();
-  const { networkId } = getStoredNetwork();
-  const classes = useStyles({ networkId });
-  const walletStore: WalletStore = stores.Wallet;
-  const accountStore: AccountStore = stores.Account;
-  const processStore: ProcessStore = stores.Process;
-  const validatorStore: ValidatorStore = stores.Validator;
-  const process: RegisterValidator | SingleCluster = processStore.getProcess;
-  process;
+    const stores = useStores();
+    const navigate = useNavigate();
+    const { networkId } = getStoredNetwork();
+    const walletStore: WalletStore = stores.Wallet;
+    const accountStore: AccountStore = stores.Account;
+    const processStore: ProcessStore = stores.Process;
+    const validatorStore: ValidatorStore = stores.Validator;
+    const classes = useStyles({ networkId });
+    const { getNextNavigation } = useValidatorRegistrationFlow(window.location.pathname);
 
-    useEffect(() => {
-        async function getNonce() {
-          return accountStore.getOwnerNonce(walletStore.accountAddress);
-        }
-         getNonce();
-    }, []);
 
+    const buttonsData: ButtonData[] = [
```

Comment:
if everything here is statically defined, move this outside of the scope of GenerateKeyShares const.

Reply:
it can't be moved now because I use there a lot of things from component

Reply:
ok I see